src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -355,24 +373,36 @@ public void scan(HttpMessage msg, String param, String value) {
         	            	// Maybe they're blocking script tags
             	            contexts2 = performAttack (msg, param, 
     			            		"<b onMouseOver=alert(1);>test</b>", context, HtmlContext.IGNORE_PARENT);
-                            if (contexts2 == null) {
-                                break;
-                            }
-    			            for (HtmlContext context2 : contexts2) {
-    			            	if ("body".equalsIgnoreCase(context2.getParentTag()) ||
-    			            			"b".equalsIgnoreCase(context2.getParentTag())) {
-    			            		// Yep, its vulnerable
-    								bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, null, param, contexts2.get(0).getTarget(), 
-    										"", contexts2.get(0).getTarget(), contexts2.get(0).getMsg());
-    								attackWorked = true;
-    								break;
-    			            	}
+            	            if (contexts2 != null) {
+	    			            for (HtmlContext context2 : contexts2) {
+	    			            	if ("body".equalsIgnoreCase(context2.getParentTag()) ||
+	    			            			"b".equalsIgnoreCase(context2.getParentTag())) {
+	    			            		// Yep, its vulnerable
+	    								bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, null, param, contexts2.get(0).getTarget(), 
+	    										"", contexts2.get(0).getTarget(), contexts2.get(0).getMsg());
+	    								attackWorked = true;
+	    								break;
+	    			            	}
+	    			            }
     			            }
+            	            if (!attackWorked) {
+                                // Try double encoded
+                                List<HtmlContext> contexts3 = performAttack (msg, param, 
+                                        URLEncoder.encode(GENERIC_SCRIPT_ALERT, UTF8), 
+                                        null, 0, true);
+                                if (contexts3 != null && contexts3.size() > 0) {
+               						bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, null, param, 
+               								URLEncoder.encode(URLEncoder.encode(contexts3.get(0).getTarget(), UTF8), UTF8), 

```

Comment:
Kind of annoying that this has to be done to display the proper "attack". I guess I could put the single encoded variant into a variable earlier, and then perform the secondary encoding here?

Reply:
Yeah, it can also be added as a constant?